platform/darwin/src/MGLShape.h

```diff
@@ -86,6 +86,15 @@ MGL_EXPORT
  */
 @property (nonatomic, copy, nullable) NSString *subtitle;
 
+/**
+ A Boolean value indicating whether the annotation is enabled. This does
+ not impact style layers created with subclasses of `MGLShape`.
+ 
+ The default value of this property is `YES`. If the value of this property is
+ `NO`, the `MGLShape` object ignores touch events and cannot be selected.
+ */
+@property (nonatomic, getter=isEnabled) BOOL enabled;
```

Comment:
Implementing this property on MGLShape prevents the developer from opting [custom MGLAnnotation classes](https://www.mapbox.com/ios-sdk/maps/examples/annotation-models/) out of selection and interactivity. At the same time, it means features that are part of a shape source or feature querying result have a meaningless `enabled` property.